### PR TITLE
feat: Add config-utils to stackable-base

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

We use `cargo cyclonedx` to place the SBOM alongside the binary, similar to what we do with the operators

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
- [ ] Changes are OpenShift compatible
- [ ] All added packages (via microdnf or otherwise) have a comment on why they are added
- [ ] Things not downloaded from Red Hat repositories should be mirrored in the Stackable repository and downloaded from there
- [ ] All packages should have (if available) signatures/hashes verified
- [ ] Add an entry to the CHANGELOG.md file
- [ ] Integration tests ran successfully
```

<details>
<summary>TIP: Running integration tests with a new product image</summary>

The image can be built and uploaded to the kind cluster with the following commands:

```shell
bake --product <product> --image-version <stackable-image-version>
kind load docker-image <image-tagged-with-the-major-version> --name=<name-of-your-test-cluster>
```

See the output of `bake` to retrieve the image tag for `<image-tagged-with-the-major-version>`.
</details>

### CODE REVIEW

File: stackable-base/Dockerfile [hunk 1]
```diff
@@ -4,7 +4,24 @@
 # https://catalog.redhat.com/software/containers/ubi9/ubi-minimal/615bd9b4075b022acc111bf5?image=6633b5136d81a5634a616c41&architecture=amd64
 
 # Manifest list digest because of multi architecture builds ( https://www.redhat.com/architect/pull-container-image#:~:text=A%20manifest%20list%20exists%20to,system%20on%20a%20specific%20architecture )
-FROM registry.access.redhat.com/ubi9/ubi-minimal@sha256:2636170dc55a0931d013014a72ae26c0c2521d4b61a28354b3e2e5369fa335a3 AS builder
+FROM registry.access.redhat.com/ubi9/ubi-minimal@sha256:2636170dc55a0931d013014a72ae26c0c2521d4b61a28354b3e2e5369fa335a3 AS product-utils-builder
+
+ARG CONFIG_UTILS_VERSION=main
```
Comment (sbernauer) [inline stackable-base/Dockerfile:9]: Marker: Once reviewed, release and use

**Review by NickLarsenNZ — DISMISSED** [review]
LGTM

**Comment by lfrancke** [comment]
Please wait with merging this.
I don't have time to review this today but have some concerns and want to spend time to look at it in more depth either later today or tomorrow.

**Comment by lfrancke** [comment]
I'll tell you what I do not like about this but I can't think of a better solution right now: More hardcoded version numbers (rust toolchain, cyclonedx, auditable) and a duplication of the other builder it'd be great if the base image could really be the base but as I said: I don't have a better idea right now.

File: stackable-base/Dockerfile [hunk 2]
```diff
@@ -4,7 +4,28 @@
...
+RUN microdnf update --assumeyes && \
+    microdnf --assumeyes install \
+    # Needed to fetch source
+    git \
+    # Needed for compilation
+    gcc && \
+    microdnf clean all
+
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --default-toolchain 1.78.0 && \
+    . "$HOME/.cargo/env" && cargo install cargo-cyclonedx@0.4.0 cargo-auditable@0.6.2
```
Comment (lfrancke) [inline stackable-base/Dockerfile:20]: I know we don't do this in the other builder but could you move all the version numbers to environment variables? That might allow us to actually use Renovate for this.

https://docs.renovatebot.com/modules/manager/regex/#regular-expression-capture-groups

**Review by lfrancke — CHANGES_REQUESTED** [review]

File: stackable-base/Dockerfile (hunk 2 above)
Comment (lfrancke) [reply stackable-base/Dockerfile:20]: I got it working without environment variables but it is definitely nicer to have env variables. Makes for a more robust regex.

Comment (sbernauer) [reply stackable-base/Dockerfile]: @lfrancke I pushed some changes, but I have no idea how to test it. Happy about any comments :)

Comment (sbernauer) [reply stackable-base/Dockerfile]: Lol I thought renovate would pick my comments up, but it seems like we still need the regex thing...

Comment (lfrancke) [reply stackable-base/Dockerfile]: Yes, we do. I didn't even expect you to include the comments, only the env vars. I'll have to check those out in detail. Sorry.

If you want to merge this fast I'm happy to approve a version without the renovate comments. If you're happy to merge later I can check those out when I have time.

Comment (sbernauer) [reply stackable-base/Dockerfile]: Ok, as I'm no expert on renovate, I removed the comments again and would be happy to address it later on :sweat: 
But I would push a commit were I do the same fix for the rust images (just for cleanup)

File: stackable-base/Dockerfile [hunk 3]
```diff
@@ -4,7 +4,32 @@
 # https://catalog.redhat.com/software/containers/ubi9/ubi-minimal/615bd9b4075b022acc111bf5?image=6633b5136d81a5634a616c41&architecture=amd64
 
 # Manifest list digest because of multi architecture builds ( https://www.redhat.com/architect/pull-container-image#:~:text=A%20manifest%20list%20exists%20to,system%20on%20a%20specific%20architecture )
-FROM registry.access.redhat.com/ubi9/ubi-minimal@sha256:a7d837b00520a32502ada85ae339e33510cdfdbc8d2ddf460cc838e12ec5fa5a AS builder
+FROM registry.access.redhat.com/ubi9/ubi-minimal@sha256:a7d837b00520a32502ada85ae339e33510cdfdbc8d2ddf460cc838e12ec5fa5a AS product-utils-builder
+
+ENV CONFIG_UTILS_VERSION=0.1.0
+# This SHOULD to be kept in sync with operator-templating and other tools to reduce build times
+ENV RUST_DEFAULT_TOOLCHAIN_VERSION=1.78.0
+ENV CARGO_CYCLONEDX_CRATE_VERSION=0.5.3
```
Comment (lfrancke) [inline stackable-base/Dockerfile:12]: Thank you for updating the versions. Unfortunately 0.5 changes the CLI and we need to adapt. Can you keep it at the old version for now?

**Review by lfrancke — CHANGES_REQUESTED** [review]

File: stackable-base/Dockerfile (hunk 3 above)
Comment (lfrancke) [reply stackable-base/Dockerfile:12]: auditable can stay updated

Comment (sbernauer) [reply stackable-base/Dockerfile:12]: I can downgrade as well, but I already adopted the CLI (nit sure if I did it correctly)

Comment (sbernauer) [reply stackable-base/Dockerfile:12]: On the new image the sbom still seems correct: `docker run -it --entrypoint bash docker.stackable.tech/stackable/stackable-base:1.0.0-stackable0.0.0-dev -c "cat /stackable/config-utils.cdx.xml | wc -l"`

Comment (sbernauer) [reply stackable-base/Dockerfile:12]: Ok, `cargo cyclonedx --output-pattern package --all --output-cdx` produced only 449 lines, so we have more than before. I leave it to you if we want to downgrade again or stick with 0.5.3

Comment (lfrancke) [reply stackable-base/Dockerfile:12]: yeah, sorry, I don't have time to look at those. Please revert the upgrade for now. There could be subtle changes in there.

File: stackable-base/Dockerfile [hunk 4]
```diff
@@ -4,7 +4,32 @@
 # https://catalog.redhat.com/software/containers/ubi9/ubi-minimal/615bd9b4075b022acc111bf5?image=6633b5136d81a5634a616c41&architecture=amd64
 
 # Manifest list digest because of multi architecture builds ( https://www.redhat.com/architect/pull-container-image#:~:text=A%20manifest%20list%20exists%20to,system%20on%20a%20specific%20architecture )
-FROM registry.access.redhat.com/ubi9/ubi-minimal@sha256:a7d837b00520a32502ada85ae339e33510cdfdbc8d2ddf460cc838e12ec5fa5a AS builder
+FROM registry.access.redhat.com/ubi9/ubi-minimal@sha256:a7d837b00520a32502ada85ae339e33510cdfdbc8d2ddf460cc838e12ec5fa5a AS product-utils-builder
+
+ENV CONFIG_UTILS_VERSION=0.1.0
+# This SHOULD to be kept in sync with operator-templating and other tools to reduce build times
```
Comment (lfrancke) [inline stackable-base/Dockerfile:10]:
```suggestion
# This SHOULD be kept in sync with operator-templating and other tools to reduce build times
```

File: ubi8-rust-builder/Dockerfile [hunk 5]
```diff
@@ -12,6 +12,12 @@ FROM registry.access.redhat.com/ubi8-minimal@sha256:5f1cd3422d5d46aea35dac80825d
 
 LABEL maintainer="Stackable GmbH"
 
+# This SHOULD to be kept in sync with operator-templating and other tools to reduce build times
```
Comment (lfrancke) [inline ubi8-rust-builder/Dockerfile:15]:
```suggestion
# This SHOULD be kept in sync with operator-templating and other tools to reduce build times
```

**Review by lfrancke — CHANGES_REQUESTED** [review]

File: stackable-base/Dockerfile (hunk 3 above)
Comment (sbernauer) [reply stackable-base/Dockerfile]: Sure, downgraded again

**Review by lfrancke — APPROVED** [review]

**Pull request merged**